server/controllers/votes/VoteController.js

```diff
@@ -0,0 +1,89 @@
+import models from '../../models';
+import {
+  ARTICLE_LIKED_MSG,
+  ARTICLE_DISLIKED_MSG,
+  ARTICLE_UNLIKED_MSG
+} from '../../helpers/responseMessages';
+
+const { ArticleVote } = models;
+/**
+ * @description The vote controller class
+ * @class
+ */
+class Vote {
+  /**
+   * @description Hanldes Like functionality
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns {object} Response
+   */
+  static async reactToAnArticle(req, res, next) {
+    try {
+      const { articleId } = req.params;
+      const userId = req.user.id;
+      const vote = req.body.vote;
+
+      const dbResult = await ArticleVote.findOrCreate(
+        { where: { userId, articleId },
+          defaults: { vote } }
+      );
+      if(!dbResult[1]) {
+        await Vote.updateArticleVote(1, userId, articleId);
+        return res.status(200).json({
+          message: vote == 1 ? ARTICLE_LIKED_MSG : ARTICLE_DISLIKED_MSG
+        });
+      }
+      return res.status(201).json({
+        message: vote == 1 ? ARTICLE_LIKED_MSG : ARTICLE_DISLIKED_MSG
+      });
+    } catch (error) {
+      return next(error);
+    }
+  }
+
+  /**
+   * @description Unlikes an article if it is already liked or disliked
+   * @param {object} req
+   * @param {object} res
+   * @param {function} next
+   * @returns {object} Response
+   */
+  static async unlikeAnAlreadyLikedOrDislikedArticle(req, res, next) {
```

Comment:
This function name needs to change.

Reply:
Okay. What name would you prefer? @VictoriaOffoma

Reply:
Come up with something concise.